testing/e2e/manifests/operator.go

```diff
@@ -37,8 +44,62 @@ var (
 
 const (
 	pathToDefaultDomainCert = "/path/to/default/domain/cert"
+	pathToDefaultDomainKey  = "/path/to/default/domain/key"
 )
 
+// generateSelfSignedCert generates a self-signed TLS certificate and private key for testing
+func generateSelfSignedCert() (certPEM, keyPEM []byte, err error) {
+	// Generate a private key
+	privateKey, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Create certificate template
+	template := x509.Certificate{
+		SerialNumber: big.NewInt(1),
+		Subject: pkix.Name{
+			Organization:  []string{"Test Organization"},
+			Country:       []string{"US"},
+			Province:      []string{""},
+			Locality:      []string{"San Francisco"},
+			StreetAddress: []string{""},
+			PostalCode:    []string{""},
+		},
+		NotBefore:   time.Now(),
+		NotAfter:    time.Now().Add(365 * 24 * time.Hour), // Valid for 1 year
+		KeyUsage:    x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature,
+		ExtKeyUsage: []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth},
+		IPAddresses: []net.IP{net.IPv4(127, 0, 0, 1), net.IPv6loopback},
+		DNSNames:    []string{"localhost", "*.example.com", "example.com"},
```

Comment:
wonder if it may be useful to feed in domain names?

Reply:
maybe in the future